File: backend/test_image_search.js
const fs = require('fs');
const FormData = require('form-data');
const axios = require('axios');

const testImageSearch = async () => {
  const imagePath = 'C:\\Users\\nethm\\Downloads\\c.jpg';

  try {
    const form = new FormData();
    const fileStream = fs.createReadStream(imagePath);
    form.append('image', fileStream);

    console.log('Sending image to http://localhost:5005/api/image-search/search...');
    const response = await axios.post(
      'http://localhost:5005/api/image-search/search',
      form,
      {
        headers: form.getHeaders(),
        timeout: 120000,  // 120 seconds to match backend
      }
    );

    console.log('Status:', response.status);
    console.log('Response:', JSON.stringify(response.data, null, 2));
  } catch (error) {
    console.error('Error:', error.message);
    if (error.response) {
      console.error('Response status:', error.response.status);
      console.error('Response data:', error.response.data);
    }
  }
};

testImageSearch();

File: test_image_search.py
import requests

image_path = r"C:\Users\nethm\Downloads\c.jpg"

try:
    with open(image_path, "rb") as f:
        files = {"image": f}  # Field name MUST be "image" to match middleware
        print("Sending request to http://localhost:5005/api/image-search/search...")
        response = requests.post("http://localhost:5005/api/image-search/search", files=files, timeout=30)
        print("Status:", response.status_code)
        print("Response:", response.json())
except Exception as e:
    print(f"Error: {e}")
